lib/chef/knife/s3_source.rb

```diff
@@ -0,0 +1,40 @@
+require 'fog'
+
+class Chef
+  class Knife
+    class S3Source
+      attr_accessor :url
+
+      def self.fetch(url)
+        source = Chef::Knife::S3Source.new
+        source.url = url
+        source.body
+      end
+
+      def body
+        bucket_obj.files.get(path).body
+      end
+
+      private
+
+      def bucket_obj
+        @bucket_obj ||= fog.directories.get(bucket)
+      end
+
+      def bucket
+        URI(@url).host
```

Comment:
If `url="https://s3.amazonaws.com/chef-azure-extn/nimisha/encrypted_data_bag_secret"`, then this returns `bucket="s3.amazonaws.com"` and `path="/chef-azure-extn/nimisha/encrypted_data_bag_secret"` , which results into failure as `s3.amazonaws.com` is not a bucket name.

This works fine if we set `bucket="chef-azure-extn"` and `path="nimisha/encrypted_data_bag_secret"`. I suppose there is an issue with parsing.